sklearn/kernel_ridge/tests/test_krc.py

```diff
@@ -0,0 +1,228 @@
+                          [0.52173913, 0.47826087]])
+    clf.dual_coef_ = dual_coef
+    assert_almost_equal(clf.predict([[-.1, -.1]]), [2])
+
+
+def test_bad_input():
+    # Test that it gives proper exception on deficient input
+    # impossible value of C
+
+    assert_raises(ValueError, KRC(alpha=-0.5).fit, X, Y)
+
+    clf = KRC()
+    Y2 = Y[:-1]  # wrong dimensions for labels
+    assert_raises(ValueError, clf.fit, X, Y2)
+
+    # error for precomputed kernels
+    clf = KRC(kernel='precomputed')
+    assert_raises(ValueError, clf.fit, X, Y)
+
+    # predict with sparse input when trained with dense
+    clf = KRC().fit(X, Y)
+    assert_raises(ValueError, clf.predict, sparse.lil_matrix(X))
+
+    Xt = np.array(X).T
+    clf.fit(np.dot(X, Xt), Y)
+    assert_raises(ValueError, clf.predict, X)
+
+    clf = KRC()
+    clf.fit(X, Y)
+    assert_raises(ValueError, clf.predict, Xt)
+
+
+def test_unicode_kernel():
+    # Test that a unicode kernel name does not cause a TypeError
+    if six.PY2:
+        # Test unicode (same as str on python3)
+        clf = KRC(kernel=u'linear')
+        clf.fit(X, Y)
+        clf.predict(T)
+
+    # Test default behavior on both versions
+    clf = KRC(kernel='linear')
+    clf.fit(X, Y)
+    clf.predict(T)
+
+
+def test_linear_elm():
+    clf = KRC(kernel='linear').fit(X, Y)
+
+    # by default should have dual_coef
+    assert_true(clf.dual_coef_.all())
+
+    assert_array_equal(clf.predict(T), true_result)
+
+
+def test_linear_elm_iris():
+    # Test that Kernel ELM with linear kernel
+    # gives plausible predictions on the iris dataset
+    # Also, test symbolic class names (classes_).
+    target = iris.target_names[iris.target]
+    clf = KRC(kernel='linear').fit(iris.data, target)
+    assert_equal(set(clf.classes_), set(iris.target_names))
+    assert_greater(np.mean(clf.predict(iris.data) == target), 0.79)
+
+
+def test_krc_clone_with_callable_kernel():
+    # create KRC with callable linear kernel, check that results are the same
+    # as with built-in linear kernel
+    krc_callable = KRC(kernel=lambda x, y: np.dot(x, y.T))
+    # clone for checking clonability with lambda functions..
+    krc_cloned = base.clone(krc_callable)
+    krc_cloned.fit(iris.data, iris.target)
+
+    krc_builtin = KRC(kernel='linear')
+    krc_builtin.fit(iris.data, iris.target)
+
+    assert_array_almost_equal(krc_cloned.dual_coef_,
+                              krc_builtin.dual_coef_)
+    assert_array_equal(krc_cloned.predict(iris.data),
+                       krc_builtin.predict(iris.data))
+    assert_array_almost_equal(krc_cloned.predict(iris.data),
+                              krc_builtin.predict(iris.data))
+
+
+def test_krc_bad_kernel():
+    krc = KRC(kernel=lambda x, y: x)
+    assert_raises(ValueError, krc.fit, X, Y)
+
+
+def test_unfitted():
+    X = "foo!"  # input validation not required when KRC is not fitted
+
+    clf = KRC()
+    assert_raises_regexp(Exception, r"not\b.*\bfitted\b",
```

Comment:
This kind of thing is tested by common tests. You do not need to.

Reply:
OK then.

Note: I just added because, due to similarity between KRC and SVC, I saw may tests from `sklearn.svm.tests.test_svm.py` could work also for KRC (and they do, as expected).